URLs from Federal Sitemaps

TheBoatyMcBoatFace opened the pull request:
**Summary**
	•	Captured URLs from all known federal government sitemaps.
	•	Organized the URLs by domain, making it easy to exempt specific sites as needed.
	•	Added sitemaps.txt, which contains the list of sitemaps crawled to generate these URLs.

**Details**
	1.	Domain-based Organization
Breaking URLs into domain-specific groups ensures flexibility in managing site-specific rules or exemptions.
	2.	sitemaps.txt
This file provides transparency into which sitemaps were crawled. It can serve as a reference for maintaining and expanding the list in the future.

**Next Steps/Questions**
	•	Are there any additional sitemaps that should be included for completeness?
	•	Let me know if any domains should be preemptively exempted.

Nintendofan885 commented:
~~FYI some domains have a sitemap file which is not linked from robots.txt~~

~~e.g.~~
* ~~https://www.climate.gov/sitemap.xml~~
* ~~https://collegescorecard.ed.gov/sitemap.xml~~
* ~~https://www.disasterassistance.gov/sitemap.xml (robots.txt is denied)~~
* ~~https://www.fisheries.noaa.gov/sitemap.xml~~

Update: didn't originally see them in the list. Sorry!

TheBoatyMcBoatFace commented:
@Nintendofan885 The code that finds the sitemaps checks for a robots.txt first and then if it doesn't find one, it checks these common sitemap paths. 

```rust
pub fn check_common_sitemap_urls(domain: &str) -> HashSet<String> {
    let client = create_client_with_timeout();
    let common_paths = [
        "sitemap.xml",
        "sitemap_index.xml",
        "sitemap",
        "sitemapindex.xml",
        "sitemap-index.xml",
        "wp-sitemap.xml",
    ];
```

I'm trying to be a good bot, but the government makes it really, really hard. 

I'm going to publish the sitemap finder code and parser in the next few days. The predecessor to my current finder [was here](https://github.com/end-of-term/eot2024/pull/8), but the good point was made that this repo is more for content and not tools.

ldko commented:
@TheBoatyMcBoatFace sorry for the delay on a response on this. I appreciate your work on pulling all the URLs from all the gov sitemaps, since it seems a thorough way of generating federal URLs. It is a bit big for reviewing purposes on GitHub, so I'm just trying to poke around at it a bit locally before merging so we will have a record of all of these URLs.

Regarding your description of the PR, I am not finding the mentioned sitemaps.txt file, and that is the one I would like to include in our Nomination Tool list because I am not going to be able to load all 56M URLs into that system.  Also, did you intend to include _report_6_FINAL.txt in this PR? It is not mentioned in the Summary and while I can guess at what it is, it doesn't match the format of other included files in the directory you added.

TheBoatyMcBoatFace commented:
Thanks @ldko 

I completely forgot to add sitemaps.txt 🤦🏻 

I just updated this PR to include it. 

If it helps, I just uploaded the directory with the URLs to google drive. It is the same content as this PR, but it may be easier to manage. I just added it, so it may take a bit for everything to upload. [Link](https://drive.google.com/drive/folders/1gLnTSa_kwLyViAe13ddvIgWtghxXvvqR?usp=sharing)

I included `_report_6_FINAL.txt` to help show what the directory contains. It isn't necessary to include.

ldko commented:
Thanks for adding the sitemaps.txt and contributing all of these URLs @TheBoatyMcBoatFace . I can see how they will be useful both for crawling and analysis of the federal domain.
